controller.rb

```diff
@@ -80,57 +80,49 @@ def account_choice # Asks user if they want to deposit/withdraw, view current ba
         puts @account.return_balance(@cust_id)
       end
       account_choice
-
-    elsif action == "deposit"
+    when "deposit"
       Dialog::space
       @acct_list = @bank.customer_accounts(@name, @pin) 
       Dialog::account_prompt 
-      account_info
+      Display::account_info(@acct_list)
       @acct_num = Dialog::pick_your_account
       if !@account_id_array.include?(@acct_num)
         Dialog::fixnum_error
-
       else
         account = bank.load_account(@acct_num)
-        amount = Dialog::deposit_amount
+        amount = -1
+        while amount < 0
+          amount = Dialog::deposit_amount
+        end
         account.deposit(amount)
         account.save_to_db
       end
       account_choice
-
-    elsif action == "withdraw"
+    when "withdraw"
       Dialog::space
       @acct_list = @bank.customer_accounts(@name, @pin) 
       Dialog::account_prompt 
-      account_info
+      Display::account_info(@acct_list)
       @acct_num = Dialog::pick_your_account
       if !@account_id_array.include?(@acct_num)
```

Comment:
When a user goes to enter their account number to either deposit or withdraw from their account, they are flamed with the error message included in the comment on line 89.